kernels/moving_average.cu

```diff
@@ -0,0 +1,69 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2021 Mark Bauer
+ *
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ *
+ */
+
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+#include <cusp/moving_average.cuh>
+#include <complex>
+
+namespace cusp {
+
+template <typename T>
+__global__ void kernel_moving_average(const T *in, T *out, int l, float s, int N) {
```

Comment:
Is the float parameter valid for int in/out?

Reply:
Yes, because to get an actual moving average from this block, scale has to be set to 1 / length (which is a float). By default, moving average just calculates a moving sum with scale = 1.